server/odc-service/src/main/java/com/oceanbase/odc/service/sqlcheck/rule/OracleAffectedRowsExceedLimit.java

```diff
@@ -0,0 +1,194 @@
+                    return -1;
+                } else {
+                    switch (this.dialectType) {
+                        case ORACLE:
+                            affectedRows = getOracleAffectedRows(originalSql, this.jdbcOperations);
+                            break;
+                        case OB_ORACLE:
+                            affectedRows = getOBOracleAffectedRows(originalSql, this.jdbcOperations);
+                            break;
+                        default:
+                            log.warn("Unsupported dialect type: {}", this.dialectType);
+                            break;
+                    }
+                }
+            } catch (Exception e) {
+                log.warn("Error in calling getAffectedRows method", e);
+                affectedRows = -1;
+            }
+        }
+        return affectedRows;
+    }
+
+    private long getOracleAffectedRows(String originalSql, JdbcOperations jdbcOperations) {
+        /**
+         * Plan hash value: 775918519
+         *
+         * ------------------------------------------------------------------- | Id | Operation | Name |
+         * Rows | Cost (%CPU)| Time | ------------------------------------------------------------------- |
+         * 0 | DELETE STATEMENT | | 9 | 3 (0)| 00:00:01 | | 1 | DELETE | T1 | | | | | 2 | TABLE ACCESS FULL|
+         * T1 | 9 | 3 (0)| 00:00:01 | -------------------------------------------------------------------
+         *
+         * Query Block Name / Object Alias (identified by operation id):
+         * -------------------------------------------------------------
+         *
+         * 1 - DEL$1 2 - DEL$1 / T1@DEL$1
+         *
+         * Column Projection Information (identified by operation id):
+         * -----------------------------------------------------------
+         *
+         * 2 - "T1".ROWID[ROWID,10]
+         */
+        String SetPlanSql =
+                "EXPLAIN PLAN SET STATEMENT_ID = '" + ODC_TEMP_EXPLAIN_STATEMENT_ID + "' FOR " + originalSql;
+        jdbcOperations.execute(SetPlanSql);
+        String getPlanSql =
+                "SELECT * FROM TABLE(DBMS_XPLAN.DISPLAY('PLAN_TABLE', '" + ODC_TEMP_EXPLAIN_STATEMENT_ID + "', 'ALL'))";
+        List<String> queryResults = jdbcOperations.query(getPlanSql, (rs, rowNum) -> rs.getString("PLAN_TABLE_OUTPUT"));
+        long estRowsValue = 0;
+        for (int rowNum = 5; rowNum < queryResults.size(); rowNum++) {
+            String resultRow = queryResults.get(rowNum);
+            estRowsValue = getEstRowsValue(resultRow);
+            if (estRowsValue != 0) {
+                break;
+            }
+        }
+        return estRowsValue;
+    }
+
+
+
+    /**
+     * OBOracle execute 'explain' statement
+     *
+     * @param originalSql target sql
+     * @param jdbcOperations jdbc Object
+     * @return affected rows
+     */
+    private long getOBOracleAffectedRows(String originalSql, JdbcOperations jdbcOperations) {
```

Comment:
Are the OBOracle and OBMySQL using the same logic? If yes, extract them into another util class.

Reply:
yes. will do